lib/plugins/aws/deploy/lib/configureStack.js

```diff
@@ -4,106 +4,163 @@ const _ = require('lodash');
+      if (!('role' in functionObject)) {
+        ret = true;
+      }
+    });
+  }
+  return ret;
+}
+/**
+ * Load the role template
+ * @returns {*} The role template
+ */
+function loadRoleTemplate() {
+  return this.serverless.utils.readFileSync(
+    path.join(this.serverless.config.serverlessPath,
+      'plugins',
+      'aws',
+      'deploy',
+      'lib',
+      'iam-role-lambda-execution-template.json')
+  );
+}
+/**
+ * Load the policy template
+ * @returns {*} The policy template
+ */
+function loadPolicyTemplate() {
+  return this.serverless.utils.readFileSync(
+    path.join(this.serverless.config.serverlessPath,
+      'plugins',
+      'aws',
+      'deploy',
+      'lib',
+      'iam-policy-lambda-execution-template.json')
+  );
+}
+/**
+ * Add the default role and policy to the CFT
+ */
+function addDefaultRoleAndPolicy() {
+  // merge in the iamRoleLambdaTemplate
+  const iamRoleLambdaExecutionTemplate = loadRoleTemplate.call(this);
 
-    if (typeof this.serverless.service.provider.iamRoleARN !== 'string') {
-      // merge in the iamRoleLambdaTemplate
-      const iamRoleLambdaExecutionTemplate = this.serverless.utils.readFileSync(
-        path.join(this.serverless.config.serverlessPath,
-          'plugins',
-          'aws',
-          'deploy',
-          'lib',
-          'iam-role-lambda-execution-template.json')
-      );
-
-      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-        iamRoleLambdaExecutionTemplate);
+  _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
+    iamRoleLambdaExecutionTemplate);
 
-      // merge in the iamPolicyLambdaTemplate
-      const iamPolicyLambdaExecutionTemplate = this.serverless.utils.readFileSync(
-        path.join(this.serverless.config.serverlessPath,
-          'plugins',
-          'aws',
-          'deploy',
-          'lib',
-          'iam-policy-lambda-execution-template.json')
-      );
+  // merge in the iamPolicyLambdaTemplate
+  const iamPolicyLambdaExecutionTemplate = loadPolicyTemplate.call(this);
 
-      // set the necessary variables for the IamPolicyLambda
-      iamPolicyLambdaExecutionTemplate
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyName = `${this.options.stage}-${this.serverless.service.service}-lambda`;
+  // set the necessary variables for the IamPolicyLambda
+  iamPolicyLambdaExecutionTemplate
+    .IamPolicyLambdaExecution
+    .Properties
+    .PolicyName = `${this.serverless.config.stage}-${this.serverless.service.service}-lambda`;
 
-      iamPolicyLambdaExecutionTemplate
-        .IamPolicyLambdaExecution
-        .Properties
-        .PolicyDocument
-        .Statement[0]
-        .Resource = `arn:aws:logs:${this.options.region}:*:*`;
+  iamPolicyLambdaExecutionTemplate
+    .IamPolicyLambdaExecution
+    .Properties
+    .PolicyDocument
+    .Statement[0]
+    .Resource = `arn:aws:logs:${this.serverless.config.region}:*:*`;
```

Comment:
We can't use `this.serverless.config.region` as it might be unset, e.g. when running `serverless deploy --noDeploy` you get an undefined in the CF template.